code/mob/input.dm

```diff
@@ -68,6 +70,42 @@
 	if (changed & (KEY_THROW|KEY_PULL|KEY_POINT|KEY_EXAMINE|KEY_BOLT|KEY_OPEN|KEY_SHOCK)) // bleh
 		src.update_cursor()
 
+
+/mob/proc/get_move_delay()
+	var/delay = max(src.movement_delay(get_step(src,src.move_dir), is_running), world.tick_lag) // don't divide by zero
+	if (move_dir & (move_dir-1))
+		delay *= DIAG_MOVE_DELAY_MULT // actual sqrt(2) unsurprisingly resulted in rounding errors
+	if (src.buckled && istype(src.buckled, /obj/stool/chair))
+		var/obj/stool/chair/C = src.buckled
+		delay += C.buckle_move_delay //GriiiiIIIND
+	for (var/obj/item/grab/G as anything in src.grabbed_by)
+		if (istype(G) && BOUNDS_DIST(src, G.assailant) > 0)
+			if (G.state > GRAB_STRONG)
+				delay += G.assailant.p_class
+
+	var/list/stepped = list()
+	for (var/obj/item/grab/G as anything in src.grabbed_by)
```

Comment:
Why are you iterating `grabbed_by` twice? I assume one of these is meant to be something else.

Reply:
spooky hacks i didn't look at hard enough. removed